var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -62,132 +63,193 @@ class Mfem(Package):
+    depends_on('metis')
+    depends_on('parmetis', when='+superlu-dist')
+    depends_on('metis@5:', when='+superlu-dist')
     depends_on('metis@5:', when='+suite-sparse ^suite-sparse@4.5:')
-    depends_on('cmake', when='^metis@5:', type='build')
+    depends_on('hypre~internal-superlu', when='+mpi')
+    depends_on('hypre@develop~internal-superlu', when='+petsc +hypre')
 
+    depends_on('sundials@2.7:+hypre', when='+sundials')
+    depends_on('suite-sparse', when='+suite-sparse')
     depends_on('superlu-dist', when='@3.2: +superlu-dist')
+    depends_on('petsc@develop', when='+petsc')
 
+    depends_on('mpfr', when='+mpfr')
+    depends_on('cmake', when='^metis@5:', type='build')
     depends_on('netcdf', when='@3.2: +netcdf')
     depends_on('zlib', when='@3.2: +netcdf')
     depends_on('hdf5', when='@3.2: +netcdf')
+    depends_on('libunwind', when='+debug')
+    depends_on('zlib', when='+gzstream')
 
     patch('mfem_ppc_build.patch', when='@3.2:3.3 arch=ppc64le')
 
     def check_variants(self, spec):
-        if '+mpi' in spec and ('+hypre' not in spec or '+metis' not in spec):
-            raise InstallError('mfem+mpi must be built with +hypre ' +
-                               'and +metis!')
-        if '+suite-sparse' in spec and ('+metis' not in spec or
-                                        '+lapack' not in spec):
-            raise InstallError('mfem+suite-sparse must be built with ' +
-                               '+metis and +lapack!')
         if 'metis@5:' in spec and '%clang' in spec and (
                 '^cmake %gcc' not in spec):
             raise InstallError('To work around CMake bug with clang, must ' +
                                'build mfem with mfem[+variants] %clang ' +
                                '^cmake %gcc to force CMake to build with gcc')
-        if '@:3.1' in spec and '+superlu-dist' in spec:
-            raise InstallError('MFEM does not support SuperLU_DIST for ' +
-                               'versions 3.1 and earlier')
-        if '@:3.1' in spec and '+netcdf' in spec:
-            raise InstallError('MFEM does not support NetCDF for versions' +
-                               '3.1 and earlier')
         return
 
+    #
+    # Note: Although MFEM does support CMake configuration, MFEM
+    # development team indicates that vanilla GNU Make is the
+    # preferred mode of configuration of MFEM and the mode most
+    # likely to be up to date in supporting *all* of MFEM's
+    # configuration options. So, don't use CMake
+    #
     def install(self, spec, prefix):
         self.check_variants(spec)
 
-        options = ['PREFIX=%s' % prefix]
-
-        if '+lapack' in spec:
-            lapack_lib = (spec['lapack'].libs + spec['blas'].libs).ld_flags  # NOQA: ignore=E501
-            options.extend([
-                'MFEM_USE_LAPACK=YES',
-                'LAPACK_OPT=-I%s' % spec['lapack'].prefix.include,
-                'LAPACK_LIB=%s' % lapack_lib])
+        yes_no = ('NO', 'YES')
```

Comment:
Really hackish? Guess I don't agree. Not sure what to say to that.

@tgamblin ...this was an idea I think I got from you in a thread on another PR I can't remember now. Can you?